### PR TITLE
[DOC] Adding CodePin Signature to Migration Workflow

### PR DESCRIPTION
**Pull request by lapel22**
Adding steps on generating and using CodePin Signature to Stage 1 and Stage 5 of Migration Workflow for 2024.2. Added Stage 5 section on validation to Migration Workflow, and updated Stage 4 to say "build" rather than "build and validate," to match the Migration Workflow outline spreadsheet.

### CODE REVIEW

**Review by jiejanezhang — COMMENTED** [review]
https://github.com/lapel22/SYCLomatic/blob/migration-workflow-codepin/docs/dev_guide/migration/migration-workflow.rst 
Shall we also add the link in the Overview section to the newly-added Stage 5?

**Review by jiejanezhang — APPROVED** [review]
LGTM.

File: docs/dev_guide/migration/migration-workflow.rst [hunk 1]
```diff
@@ -86,6 +86,22 @@ in the migration results:
 For detailed information about dialect differences between Clang and nvcc, refer to llvm.org's
 `Compiling CUDA with clang <https://www.llvm.org/docs/CompileCudaWithLLVM.html>`_ page.
 
+Run CodePin to Capture Application Signature
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CodePin is feature of |tool_name| that helps reduce the effort of debugging inconsistencies in runtime behavior. CodePin generates reports
```
Comment (AndyCHHuang) [inline docs/dev_guide/migration/migration-workflow.rst:92]: CodePin is a feature

File: docs/dev_guide/migration/migration-workflow.rst [hunk 2]
```diff
@@ -86,6 +86,22 @@ in the migration results:
...
 `Compiling CUDA with clang <https://www.llvm.org/docs/CompileCudaWithLLVM.html>`_ page.
 
+Run CodePin to Capture Application Signature
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CodePin is feature of |tool_name| that helps reduce the effort of debugging inconsistencies in runtime behavior. CodePin generates reports
+from the CUDA and SYCL programs that, when compared, can help identify the
+source of divergent runtime behavior.
+
+Use the CodePin tool before migration in order to capture the project signature.
```
Comment (AndyCHHuang) [inline docs/dev_guide/migration/migration-workflow.rst:96]: Enable the CodePin tool during the migration

File: docs/dev_guide/migration/migration-workflow.rst [hunk 3]
```diff
@@ -354,6 +369,20 @@ in the Codeplay plugin documentation:
...
+Stage 5: Validate the New SYCL Application
+------------------------------------------
+
+After you have built your converted code, validate your new SYCL application to
+check for correct functionality after migration.
+
+Use CodePIN to Validate Migrated Code
+*************************************
+
+Use the CodePIN signature (captured in Stage 1) to validate your migrated code.
```
Comment (AndyCHHuang) [inline docs/dev_guide/migration/migration-workflow.rst:381]: If the CodePin feature has been enabled during the migration time,
project signature will be logged during the execution time.
The signature contains the data value of each execution checkpoint, which can be verified manually or with an auto-analysis tool.

**Pull request closed without merging**